tools/monitoring/cloudmail/mail.go

```diff
@@ -0,0 +1,222 @@
+	domain := &mailpb.Domain{
+		ProjectDomain: true,
+		DomainName:    "",
+	}
+	req := &mailpb.CreateDomainRequest{
+		Parent: fmt.Sprintf("projects/%s", projectID),
+		Region: region,
+		Domain: domain,
+	}
+	resp, err := c.CloudMailClient.CreateDomain(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	c.DomainID = strings.Replace(resp.GetName(), fmt.Sprintf("regions/%s/domains/", region), "", 1)
+	c.DomainName = resp.GetDomainName()
+
+	fmt.Printf("Domain created.\n")
+	fmt.Printf("Domain name: %s\n", c.DomainName)
+	fmt.Printf("Domain region: %s\n", region)
+	fmt.Printf("Domain ID: %s\n", c.DomainID)
+
+	return nil
+}
+
+// CreateAddressSet enables email addresses under the domain
+func (c *MailClient) CreateAddressSet(ctx context.Context) error {
+	addressSet := &mailpb.AddressSet{
+		AddressPatterns: []string{addressPattern},
+	}
+	req := &mailpb.CreateAddressSetRequest{
+		Parent:       fmt.Sprintf("regions/%s/domains/%s", region, c.DomainID),
+		AddressSetId: addressSetID,
+		AddressSet:   addressSet,
+	}
+
+	if _, err := c.CloudMailClient.CreateAddressSet(ctx, req); err != nil {
+		return err
+	}
+	fmt.Printf("Address set created.\nAddress set ID: %s\n", addressSetID)
+	return nil
+}
+
+// CreateSenderDomain configures the sender email
+func (c *MailClient) CreateSenderDomain(ctx context.Context) error {
+	addressSetPath := fmt.Sprintf("regions/%s/domains/%s/addressSets/%s", region, c.DomainID, addressSetID)
+	sender := &mailpb.Sender{
+		DefaultEnvelopeFromAuthority: addressSetPath,
+		DefaultHeaderFromAuthority:   addressSetPath,
+	}
+	req := &mailpb.CreateSenderRequest{
+		Parent:   fmt.Sprintf("projects/%s", projectID),
+		Region:   region,
+		SenderId: senderID,
+		Sender:   sender,
+	}
+
+	if _, err := c.CloudMailClient.CreateSender(ctx, req); err != nil {
+		return err
+	}
+	fmt.Printf("Sender created.\nSender ID: %s\n", senderID)
+	return nil
+}
+
+// CreateAndApplyReceiptRuleDrop configures the bounce message behaviour to do nothing when an email cannot be delivered
+func (c *MailClient) CreateAndApplyReceiptRuleDrop(ctx context.Context) error {
+	const matchMode = "PREFIX"
+
+	cloudMailClient := c.CloudMailClient
+
+	envelopeToPattern := &mailpb.ReceiptRule_Pattern{
+		Pattern:   addressPattern,
+		MatchMode: mailpb.ReceiptRule_Pattern_MatchMode(mailpb.ReceiptRule_Pattern_MatchMode_value[matchMode]),
+	}
+	dropAction := &mailpb.ReceiptAction_Drop{
+		Drop: &mailpb.DropAction{},
+	}
+	action := &mailpb.ReceiptAction{
+		Action: dropAction,
+	}
+	receiptRule := &mailpb.ReceiptRule{
+		EnvelopeToPatterns: []*mailpb.ReceiptRule_Pattern{envelopeToPattern},
+		Action:             action,
+	}
+	createReceiptRuleReq := &mailpb.CreateReceiptRuleRequest{
+		Parent:      fmt.Sprintf("regions/%s/domains/%s", region, c.DomainID),
+		RuleId:      receiptRuleID,
+		ReceiptRule: receiptRule,
+	}
+
+	if _, err := cloudMailClient.CreateReceiptRule(ctx, createReceiptRuleReq); err != nil {
```

Comment:
should be able to use `c.CreateReceiptRule()`

Reply:
Done